src/sipmanalyze/estimate.py

```diff
@@ -0,0 +1,249 @@
+"""
+
+estimate.py
+
+Functions to help with estimation of low-light parameters.
+
+"""
+
+import matplotlib.pyplot as plt
+import sipmanalyze.formats as forms
+import awkward as ak
+import numpy as np
+
+import sipmanalyze.plotting as plot
+import hist
+
+from scipy.signal import find_peaks
+import scipy.integrate as integrate
+from scipy.optimize import curve_fit
+from scipy.special import factorial
+
+
+#- Use this function to import the lowlight response data from a root file
+#- file_location is a string giving the location of the imported file
+#- import_data_results is a dictionary with the results of the import used in later functions:
+#  area: raw data
+#  amax: max value in area array
+#  amin: min value in area array
+#  nbins: number of bins in histogram
+#  bin_width: width of bin in mv-ns
+#  n: number of counts in each bin
+#  axis: his.axis object
+#  h: histogram object filled with data
+def import_data(file_location):
+  #get data
+  container = forms.waveform.waveform_container.from_root(file_location)
+  tint = container.settings.timeintervals
+  vadc = container.settings.adc_val
+  area = ak.sum(container.waveforms[:, 5:30], axis=-1) * tint * vadc * -1
+  amin, amax = ak.min(area), ak.max(area)
+  nbins = int((amax - amin) / tint / vadc / 4)
+  bin_width=(amax-amin)/nbins
+    
+  #bin data in histogram
+  axis=hist.axis.Regular(nbins, amin, amax, name='r')
+  h = hist.Hist(axis)
+  h.fill(area)
+  
+  #save data in dict
+  bins, _= h.to_numpy()
+  import_data_results={"area":area,"amax":amax,"amin":amin,"nbins":nbins,"bin_width":bin_width, "n":bins, "axis":axis,"h":h}
+  return import_data_results
+
+#- use this function to plot the results of importing the data
+#- import_data_results is the dictionary of results returned by import_data
+def plot_import_data(import_data_results):
+  #import histogram data
+  h=import_data_results["h"]
+  
+  #plot histogram data
+  fig,ax = plot.make_simple_figure()
+  plot.plot_data1d(ax=ax, data=h, label='Data', histtype='fill')
+  plot.add_std_label(ax=ax, label='Preliminary', rlabel='Low Light Response')
+  ax.set_xlabel('Readout [mV-ns]')
+  ax.set_ylabel('Number of events')
+  ax.legend()
+
```

Comment:
Return the matplotlib `fig,ax` pair, this allows for additional styling to be done by the user.